### PR TITLE
feature/double-auth-match

### LINKED ISSUE #1279: Combine the calls for matching rules role and regdate
Since the rules are now split between auth and admin it is now possible to combine the calls to pf::authentication::match for the role and regdate actions.

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Avoid the calling pf::authentication::match twice
## Impacts

Everything
## Issue

Fixes #1279
## Delete branch after merge

NO

### CODE REVIEW

File: lib/pf/api.pm [hunk 1]
```diff
@@ -1040,9 +1040,14 @@ sub dynamic_register_node : Public {
...
-    #Compute autoreg if we use autoreg
-    my $value = &pf::authentication::match([@sources], $params, $Actions::SET_UNREG_DATE);
+    my $matched = pf::authentication::match2([@sources], $params);
+    unless ($matched) {
+        $logger->warn("Did not find any actions to match");
+        return;
+    }
+    my $values = $matched->{values};
+    my $role = $values->{$Actions::SET_ROLE};
+    my $value = $values->{$Actions::SET_UNREG_DATE};
```
Comment (julsemaan) [inline lib/pf/api.pm]: This would be clearer if the variable was called $unregdate

**Comment by julsemaan** [comment]
This code looks pretty good given the code paths it plays into.

Would require testing before merge but other than the minor detail above, it's good.

**Comment by jrouzierinverse** [comment]
Rebase please review

**Comment by extrafu** [comment]
@julsemaan Review please!

**Pull request merged**